packages/actor-init-query/lib/QueryEngineBase.ts

```diff
@@ -249,6 +249,11 @@ implements IQueryEngine<QueryContext, QueryStringContextInner, QueryAlgebraConte
       };
     }
 
+    // Invalidate caches if cache argument is set to false
+    if (actionContext.get(KeysInitQuery.noCache)) {
```

Comment:
While reading this code, I'm realizing that this will only invalidate cache for the next query execution.

If a query execution was executed before without noCache, then the cache will still be filled when executing this query, and will only be invalidated afterwards.

I think we may want to move this block so that it's the very first thing that happens before actual query execution.

What do you think?

Reply:
I think this is a good suggestion. In light of the case where the user wants their execution to not be influenced by what the cache contains it makes sense to clear the cache before execution. This should have an impact on performance, I'll test it out.

In the case where the user wants to avoid memory overhead it makes sense to also clear the cache after query execution. So I would look at the impact on performance of clearing the caches before and after.

And there's the case where the user wants one of their queries to have no influence on the cache, also leaving them intact. In that case clearing the caches works against what the user intends. Do you think this use case is common?

Reply:
Maybe clearing cache before and after exec is indeed what we want, if performance impact is not too high.

Reply:
Impact on performance seems very limited when clearing the cache before and after execution, so let's continue with this solution.
What about the use case of wanting to do a query that leaves cache state intact?

Reply:
> What about the use case of wanting to do a query that leaves cache state intact?

I would leave this out of scope for this PR.